evdev/uinput.py

```diff
@@ -281,11 +283,75 @@ def _verify(self):
             msg = 'uinput device name must not be longer than {} characters'
             raise UInputError(msg.format(_uinput.maxnamelen))
 
-    def _find_device(self):
+    def _find_device(self, fd):
+        '''
+        Tries to find the device node. Will delegate this task to one of
+        several platform-specific functions.
+        '''
+        if platform.system() == 'Linux':
+            try:
+                sysname = _uinput.get_sysname(fd)
+                return self._find_device_linux(sysname)
+            except OSError:
+                # UI_GET_SYSNAME returned an error code. We're likely dealing with
+                # an old kernel. Guess the device based on the filesystem.
+                pass
+
+        # If we're not running or Linux or the above method fails for any reason,
+        # use the generic fallback method.
+        return self._find_device_fallback()
+
+    def _find_device_linux(self, sysname):
+        '''
+        Tries to find the device node when running on Linux.
+        '''
+
+        syspath = f'/sys/devices/virtual/input/{sysname}'
+
+        # The sysfs entry for event devices should contain exactly one folder
+        # whose name matches the format "event[0-9]+". It is then assumed that
+        # the device node in /dev/input uses the same name.
+        regex = re.compile('event[0-9]+')
+        for entry in os.listdir(syspath):
```

Comment:
What about wrapping `for entry in os.listdir(syspath):` in try/except as well? You weren't sure if `syspath` is populated immediately on old systems, has this changed?

Reply:
See my comments below for a potentially better alternative